File: backend/src/main/java/com/demo/aiknowledge/entity/ProductSku.java
package com.demo.aiknowledge.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Map;

@Data
@TableName(value = "product_sku", autoResultMap = true)
public class ProductSku {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long productId;
    private String skuCode;
    @com.baomidou.mybatisplus.annotation.TableField(typeHandler = JacksonTypeHandler.class)
    private Map<String, String> properties;
    private BigDecimal price;
    private Integer stock;
    private Boolean isDefault;
    private LocalDateTime createTime;
}

File: android/app/src/main/java/com/evanyao/shopagent/data/model/ProductSku.kt
package com.evanyao.shopagent.data.model

import com.google.gson.annotations.SerializedName
import java.math.BigDecimal

data class ProductSku(
    val id: Long,
    @SerializedName(value = "productId", alternate = ["product_id"])
    val productId: Long,
    @SerializedName(value = "skuCode", alternate = ["sku_code"])
    val skuCode: String? = null,
    val properties: Map<String, String>? = null,
    val price: BigDecimal,
    val stock: Int = 0,
    @SerializedName(value = "isDefault", alternate = ["is_default"])
    val isDefault: Boolean = false,
    val createTime: String? = null
) {
    // 显示规格文本
    val propertiesText: String
        get() {
            if (properties.isNullOrEmpty()) return ""
            return properties.entries.joinToString(" ") { "${it.key}: ${it.value}" }
        }
}
